R/createAttributePlan.R

```diff
@@ -0,0 +1,190 @@
+##                    Create attribute plan                    --
+##---------------------------------------------------------------
+#' @param jaspResults {object} Object that will contain all results from the analysis and connect it to the output.
+#' @param dataset {object} (optional) tabular data, if available for the analysis.
+#' @param options {list} A named list of interface options selected by the user.
+#' @seealso .findPlan()
+##---------------------------------------------------------------
+CreateAttributePlan <- function(jaspResults, dataset = NULL, options, ...) {
+  # Constraints to create a plan
+  risk_vars <- c("aql", "prod_risk", "rql", "cons_risk")
+  # Quality levels
+  pd_vars <- c("pd_lower", "pd_upper", "pd_step")
+  depend_vars <- c(pd_vars, risk_vars, "lotSize", "distribution")
+
+  # Check if the container already exists. Create it if it doesn't.
+  if (is.null(jaspResults[["createContainer"]]) || jaspResults[["createContainer"]]$getError()) {
+    createContainer <- createJaspContainer(title = "")
+    createContainer$dependOn(depend_vars)
+    jaspResults[["createContainer"]] <- createContainer
+  } else {
+    createContainer <- jaspResults[["createContainer"]]
+  }
+
+  # Plan table outline
+  plan_table <- createJaspTable(title = "Generated Sampling Plan")
+  plan_table$addColumnInfo(name = "col_1", title = "", type = "string")
+  plan_table$addColumnInfo(name = "col_2", title = "Value", type = "integer")
+  plan_table[["col_1"]] <- c("Sample size", "Acceptance number")
+  plan_table$position <- 1
+  createContainer[["findPlanTable"]] <- plan_table
+
+  # Probability table outline
+  prob_table <- createJaspTable(title = "Acceptance probabilities at AQL and RQL")
+  prob_table$addColumnInfo(name = "col_1", title = "", type = "string")
+  prob_table$addColumnInfo(name = "col_2", title = "Proportion Non-conforming", type = "number")
+  prob_table$addColumnInfo(name = "col_3", title = "Acceptance Probability", type = "number")
+  prob_table$addColumnInfo(name = "col_4", title = "Rejection Probability", type = "number")
+  prob_table[["col_1"]] <- c("AQL", "RQL")
+  prob_table$position <- 2
+  createContainer[["findProbTable"]] <- prob_table
+  
+  # Error handling for hypergeometric distribution
+  aql <- round(options$aql, 3)
+  rql <- round(options$rql, 3)
+  checkHypergeom(createContainer, pd_vars, options, type="", aql, rql)
+  if (createContainer$getError()) {
+    return ()
+  }
+  
+  # Error handling for AQL/RQL
+  if (aql >= rql) {
+    createContainer$setError(sprintf("AQL (Acceptable Quality Level) value should be lower than RQL (Rejectable Quality Level) value."))
+    return ()
+  }
+
+  # Error handling for Producer's and Consumer's Risk
+  pa_prod <- round((1 - options$prod_risk), 3)
+  pa_cons <- round(options$cons_risk, 3)
+  if (pa_prod <= pa_cons) {
+    createContainer$setError(sprintf("1 - α (Producer's risk) has to be greater than β (consumer's risk)."))
```

Comment:
```suggestion
    createContainer$setError(gettext("1 - α (Producer's risk) has to be greater than β (consumer's risk)."))
```
also, I think you need to use the Unicode representations for α and β (resp.`\u03B1` and `\u03B2`) otherwise the sentence cannot be translated properly. @boutinb can you confirm this?

Reply:
I've updated it everywhere in the code, including for QML and help pages. :)